docusaurus/docs/Android/04-compose/04-message-components/07-selected-reactions-menu.mdx

```diff
@@ -0,0 +1,176 @@
+# SelectedReactionsMenu
+
+The `SelectedReactionsMenu` is a component that displays a list of message reactions with the users who left them. If a message already contains reactions, the user can tap on the section with reactions to show the component.
+
+This is a **stateless component** that you can easily add to your UI if you're building a custom Messages screen. The component shows the following UI:
+
+* **Reaction options**: The top part that shows a list of reactions the user can use to react to the message.
+* **User reactions**: The main part that shows a list of users with their reactions left for the message.
+
+Let's check how to use the component!
+
+## Usage
+
+If you're using the [`MessagesScreen`](./01-messages-screen.mdx), the `SelectedReactionsMenu` is already set up for you. To use the `SelectedReactionsMenu` in your custom screens, simply add it to your UI, like so:
+
+```kotlin
+// The rest of your UI
+if (selectedMessageState is SelectedMessageReactionsState) {
+    val selectedMessage = selectedMessageState.message
+    SelectedReactionsMenu(
+        modifier = Modifier.align(Alignment.BottomCenter),
+        // The currently logged-in user
+        currentUser = user,
+        // The message whose reactions you selected
+        message = selectedMessage,
+        onMessageAction = { action ->
+            // Handle message action
+        },
+        onDismiss = { 
+            // Handle dismiss
+        }
+    )
+}
+
+```
+
+As you can see, showing the menu is very simple. If the `selectedMessageState` is an instance of `SelectedMessageReactionsState`, you pass in the the currently logged in user, as well as the selected message. The reactions you show are taken from the [`ChatTheme`](../02-general-customization/01-chat-theme.mdx) component and everything else required to show the component is taken care of internally.
+
+The aforementioned code produces the following UI:
+
+![Default SelectedReactionsMenu component](../../assets/compose_default_selected_reactions_menu.png)
+
+The component shows reactions from the [`ChatTheme`](../02-general-customization/01-chat-theme.mdx) on the top, followed by a list of message reactions.
+
+To dismiss the component just tap outside of the dialog or press the system back button.
+
+## Handling Actions
+
+Here are the actions exposed by the `SelectedReactionsMenu` component:
+
+```kotlin
+@Composable
+fun SelectedReactionsMenu(
+    ..., // State
+    onMessageAction: (MessageAction) -> Unit,
+    onDismiss: () -> Unit = {},
+)
+```
+
+* `onMessageAction`: Handler for the user performing an action such as leaving a reaction.
```

Comment:
I personally don't like this construct, we seem to be using it in quite a few places.

I think sentences such as "Handler used when..." imply causality better than "Handler for the user..."